homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -264,6 +269,7 @@ async def async_step_ssdp(self, discovery_info: ssdp.SsdpServiceInfo) -> FlowRes
 
         if (
             existing_entry
+            and existing_entry.state != ConfigEntryState.LOADED
```

Comment:
This is actually not correct. Just because it's loaded, doesn't mean it's functioning.

Reply:
It only means it probably WAS functioning at some point but the IP may have changed later and the polling is now failing

Reply:
One alternative would be to use something like https://github.com/home-assistant/core/blob/dev/homeassistant/components/samsungtv/__init__.py#L120